File: Data Structures & Algorithms/valid-sudoku/submission-0.java
class Solution {
    public int getBox (int row, int col){
        return 3 * (int)(row / 3) + (int)(col / 3);
    }
    public boolean isValidSudoku(char[][] board) {
        List<Set<Character>> boxes = new ArrayList<>();
        List<Set<Character>> rows = new ArrayList<>();
        List<Set<Character>> cols = new ArrayList<>();

        for (int i = 0; i < 9; i++) {
            boxes.add(new HashSet<>());
            rows.add(new HashSet<>());
            cols.add(new HashSet<>());
        }

        for(int row = 0; row < 9; row++){
            for(int col = 0; col < 9; col++){
                char val = board[row][col];
                if (val == '.')
                    continue;
                int boxIndex = getBox(row, col);
                if(boxes.get(boxIndex).contains(val) || rows.get(row).contains(val) || cols.get(col).contains(val))
                    return false;
                boxes.get(boxIndex).add(val);
                rows.get(row).add(val);
                cols.get(col).add(val);
            }
        }

        return true;
    }
}

File: Data Structures & Algorithms/search-for-word/submission-0.py
class Solution:
    def exist(self, board: List[List[str]], word: str) -> bool:
        directions = ((-1, 0), (1, 0), (0, -1), (0, 1))
        n = len(word)
        rows, cols = len(board), len(board[0])
        visited = [[False] * cols for _ in range(rows)]
        exists = False

        def dfs (index, row, col):
            nonlocal exists
            if exists: return
            if index == n - 1:
                exists = True
                return
            visited[row][col] = True
            for d_row, d_col in directions:
                n_row, n_col = row + d_row, col + d_col
                if 0 <= n_row < rows and 0 <= n_col < cols and word[index + 1] == board[n_row][n_col] and not visited[n_row][n_col]:
                    dfs(index + 1, n_row, n_col)
            visited[row][col] = False

        for row in range(rows):
            for col in range(cols):
                if board[row][col] == word[0]:
                    dfs(0, row, col)
                    if exists: return True
        return False
